testsuite/utils.py

```diff
@@ -138,9 +138,6 @@ def asdict(obj) -> dict[str, JSONValues]:
 
 
 def _asdict_recurse(obj):
-    if hasattr(obj, "asdict"):
```

Comment:
I think changes to this  function can be reverted

Reply:
removed